src/charm.py

```diff
@@ -188,9 +373,58 @@ def _restart_postgresql_service(self) -> None:
         self.unit.status = ActiveStatus()
 
     @property
-    def _pod_ip(self) -> str:
-        """Current pod ip."""
-        return self.model.get_binding(PEER).network.bind_address
+    def _patroni(self):
+        """Returns an instance of the Patroni object."""
+        return Patroni(
+            self._endpoint,
+            self._endpoints,
+            self._namespace,
+            self._unit_ip,
+            self._storage_path,
+        )
+
+    @property
+    def _unit_ip(self) -> str:
+        """Current unit ip."""
+        return str(self.model.get_binding(PEER).network.bind_address)
```

Comment:
Is there a reason to be worrying about IPs on Kubernetes when you can use the cluster DNS? I.e. `postgresql-k8s-0.postgresql-k8s-endpoints.model_name.svc.cluster.local`?

Reply:
I fixed this now, Jon. I previously had an issue that forced the presence of the IP address only in the Patroni bootstrap process (removing the IP later and using only hostnames). Now this is also fixed (127.0.0.1 is used in the bootstrap process, keeping the change to hostnames after that).